Color code thumbs up/down in evaluation

BTWS2 opened the pull request:
This pull request closes #2715.

I don't have strong opinions on the order of thumbs up or down, but I'm in favor of color coding (thumbs up = green, thumbs down = red) to nudge towards the correct button as @rien suggested. I can switch the order if needed:
https://github.com/dodona-edu/dodona/blob/develop/app/views/feedbacks/_feedback_actions.html.erb
https://github.com/dodona-edu/dodona/blob/develop/app/views/feedbacks/_score_actions.html.erb

![image](https://user-images.githubusercontent.com/56451049/158674135-eaccbe81-cb2f-490e-8773-b28b90eb8ded.png)
![image](https://user-images.githubusercontent.com/56451049/158674215-407c1867-fafa-4554-b4a5-2ac78038557e.png)
![image](https://user-images.githubusercontent.com/56451049/158674676-a20639be-f5dd-4b09-bb5c-a0f7914c7103.png)
![image](https://user-images.githubusercontent.com/56451049/158674720-28687053-b754-43d2-b952-68d38fae3b84.png)

jorg-vr commented:
![Peek 2022-03-17 10-54](https://user-images.githubusercontent.com/21177904/158784052-e417e6c8-d982-42ad-8aed-8c1ba21a6728.gif)

I prefer to update the icon color instead of background color. (But this is just a matter of taste)

Overall I approve of the change

BTWS2 commented:
My goal with this approach is to add a visual confirmation the teacher pressed the right button. This shows the color after pressing the button. When the teacher hovers over the wrong button, a glimpse of the wrong color is shown with the previous approach. I agree with @jorg-vr that changing the color of the icon looks cleaner.

It's based on https://css-tricks.com/different-transitions-for-hover-on-hover-off/.

https://user-images.githubusercontent.com/56451049/158904550-ef138b63-afcf-4de2-8544-4e7b0a821dd7.mp4

bmesuere commented:
Why not combine both behaviours? Color on mouse over + a slight delay after clicking?

BTWS2 commented:
@bmesuere A color change on mouse over (without animation) and visual feedback after clicking would indeed be better. If I had to choose I prefer the visual feedback after clicking, but both is better.

BTWS2 commented:
Changing the `color` property in both states wasn't possible without javascript . This is the new proposal using pure CSS: small color change on hover and clicking shows more green/red for clear visual feedback.

https://user-images.githubusercontent.com/56451049/159768808-a9711bd5-5807-4b4b-86ac-258c63ab0235.mp4

chvp commented on app/assets/stylesheets/components/btn.css.scss:
```diff
@@ -291,3 +291,35 @@ form.button_to {
   @include shadow-z4;
 }
+
+.single-zero-button, #zero-button{
```
```suggestion
.single-zero-button, #zero-button {
```
(Same for the other braces.)

chvp commented on app/assets/stylesheets/components/btn.css.scss:
```diff
@@ -291,3 +291,35 @@ form.button_to {
+  background-color: $brand-success;
+  color: unset;
+  transition: background-color 0s;
+}
```
Please set up your editor to use the `.editorconfig` file in the root of the repository. That should make it so a final newline is inserted.

chvp reviewed (changes requested):
The background color is still quite harsh, I think. The end of the animation also looks a little strange to me. Maybe a fade-out would make it less jarring?

BTWS2 commented:
I changed the background to lighter shade of the color. The end of the animation is cut off at the end, because the button is updated when the loading is done (visible at the end of the screen recording of dark mode below). I made the animation shorter and used `ease-out` to mitigate this partly. I added a space before the brackets and added a newline at the end.

At first I chose a lighter color (`$danger-100`), but it resulted in even more harsh colors in dark mode (e.g. `$danger-100: $red-800 !global;`), so I used opacity instead.

Light mode:

https://user-images.githubusercontent.com/56451049/161397648-62fee4d5-b0c6-433c-83c6-1bed54ca3b30.mp4

Dark mode:

https://user-images.githubusercontent.com/56451049/161397654-fadb9aa3-5833-4d10-8022-500b1ab24b8f.mp4
